### PR TITLE
[Magiclysm] Add dabbler hobbies for the people that knew 1 spell

### PR DESCRIPTION
**Pull request by Standing-Storm**
#### Summary
Mods "[Magiclysm] Add dabbler hobbies for people knew 1 spell"

#### Purpose of change

Magiclysm lore indicates that 10% of the world's population knew at least 1 spell before the Cataclysm, and most of them only knew that one, but there was no way to start as one of those people (except for one of the magical professions who only knew 1 spell)

#### Describe the solution

This PR adds some hobbies, each of which teaches you only one spell. These are only for classless spells of low Difficulty: Freshen Up, Crystallize Mana, Magical Light, Aura of Protection, Sound Bomb, or Knowing the Day and the Hour.  There's probably plenty of other hobbies that could be added but I also don't want to add like 100 possible hobbies, each of which has 1 spell, either.

Edit: Actually looking through Druid spells there’s only one spell that fits the criteria I used here (Difficulty 3 or less and useful in a daily life that does not involve killing people), so maybe I’ll do another PR with classed dabbler spells. 

It also adds 3 hobbies for Spellcraft, matching the Basic/Intermediate/Advanced hobbies for other skills.

#### Testing

Took the hobbies, the appropriate spells all appeared.

#### Additional context

It's a little weird that character creation describes Spellcraft 4 as "advanced" when the Game Balance document says that's "Proficient hobbyist."

### CODE REVIEW

**Comment by ashGlaw** [comment]
More ways to start with spells gets 1000% approval from me. I think the dabbler hobbies for class spells are a great idea too.

File: data/mods/Magiclysm/hobbies.json [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  },
+  {
+    "type": "profession",
+    "subtype": "hobby",
+    "id": "dabbler_crystallize_mana",
+    "name": "Dabbler (Crystallize Mana)",
+    "description": "You were one of the hundreds of millions of people who knew a single spell.  In your case, Crystallize Mana.  Sure, it ruined your social life and you slept for 12 hours a day, but you made a tidy sum of money producing mana crystals for mages who needed them.  Too bad all that money is useless now.",
+    "points": 4,
+    "spells": [ { "id": "crystallize_mana", "level": 5 } ]
+  },
```
Comment (b3brodie) [inline data/mods/Magiclysm/hobbies.json:46]: Crystallize Mana hobby appears twice.

File: data/mods/Magiclysm/hobbies.json [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  {
+    "type": "profession",
+    "subtype": "hobby",
+    "id": "dabbler_magical_light",
+    "name": "Dabbler (Magical Light)",
+    "description": "You were one of the hundreds of millions of people who knew a single spell.  In your case, Magical Light.  It was mostly a party trick, to be honest, and learning to showed up how much work trying to learn real magic actually was.  But the nights are dark enough nowadays that you're glad you put in the effect.",
```
Comment (b3brodie) [inline data/mods/Magiclysm/hobbies.json:61]:
```suggestion
    "description": "You were one of the hundreds of millions of people who knew a single spell.  In your case, Magical Light.  It was mostly a party trick, to be honest, and learning it showed how much work trying to learn real magic actually was.  But the nights are dark enough nowadays that you're glad you put in the effort.",
```

**Comment by Standing-Storm** [comment]
> Would it make sense for you to select a hobby the grants one spell and then for the game to ask you to choose which one of the spells you want from a list similar to the way that starting with martial arts does? That way you could expand the Options, very easily without glutting the UI

This would be pretty nice but unfortunately it seems impossible--those use a hardcoded `initial_ma_styles` mutation parameter to provide the popup and there's nothing similar for spells. 

On the other hand, it looks like there's only about ~22 spells that fit the criteria I'm using, +1 one that doesn't but I'm including it for the sickos out there ("Carrion Feast"), so while that's a sizeable amount it's not going to overwhelm the backgrounds menu.

File: data/mods/Magiclysm/hobbies.json [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    "description": "You were one of the hundreds of millions of people who knew a single spell.  In your case, Sound Bomb.  You mostly used it for jokes, to play pranks on your friends and annoy the neighborhood dogs, but it's proven its worth now at the end of the world.",
+    "points": 4,
+    "spells": [ { "id": "sound_bomb", "level": 5 } ]
+  },
+  {
+    "type": "profession",
+    "subtype": "hobby",
+    "id": "dabbler_watch_spell",
+    "name": "Dabbler (Knowing the Day and the Hour)",
+    "description": "You were one of the hundreds of millions of people who knew a single spell.  In your case, Knowing the Day and the Hour.  Your non-mage friends were always asking you why you bothered learning what was widely considered to be the least useful spell in the world, and sometimes you weren't sure yourself.  Even now that the network is done, you're still not entirely sure how important it is to know that it's exactly 6:12 a.m. vs. just \"sunrise\".  But if it ever is important, you'll know it.",
```
Comment (Maleclypse) [inline data/mods/Magiclysm/hobbies.json:79]:
```suggestion
    "description": "You were one of the hundreds of millions of people who knew a single spell.  In your case, Knowing the Day and the Hour.  Your non-mage friends were always asking you why you bothered learning what was widely considered to be the least useful spell in the world, and sometimes you weren't sure yourself.  Even now that the network is down, you're still not entirely sure how important it is to know that it's exactly 6:12 a.m. vs. just \"sunrise\".  But if it ever is important, you'll know it.",
```

**Pull request merged**